### ISSUE TITLE
Vulkan: Manually emitted particles hide and freeze when the first one expires

### ISSUE THREAD

**Issue by FreegleBarr** [godotengine/godot#56886]
### Godot version

v4.0.dev.20220105.official.8f6cba3a4 (have also tested in a unrelatedly modified 846c14eee )

### System information

Ubuntu 20.04

### Issue description

If one emits multiple particles through GDScript, all remaining particles will become invisible a small amount of time after the first particle of the series expires. If a new particle P is emitted, the invisible particles reappear at the exact same spot they hid and continue their trajectory, independent of time past. These particles will not become hidden, but particles emitted after P will hide/freeze again after P expires.

### Steps to reproduce

Create a slow-moving GPUParticle3D, and set up some manual emission trigger through GDScript, and emit multiple particles. Make sure you don't accidentally try to emit particles during the bug.

In the reproduction project, set the particle speed to 0.5 and press a directional key, to emit a particle, 8-9 times in quick succession and stop before the first particle disappears. You'll notice that all particles but the first will incorrectly disappear at around the same time. If you press a directional key again the particles that disappeared incorrectly will reappear for a short while before disappearing again, this time correctly.

### Minimal reproduction project

[Particle_emission.zip](https://github.com/godotengine/godot/files/7885618/Particle_emission.zip)

**Comment by GanidhuAbey** [comment]
yea the problem was that the `inactive_time` of the particles weren't being reset when a new particle was spawned

I'll make a pull request sometime tonight

**Comment by reduz** [comment]
I cant see anything on the reproduction project nor I properly understand the issue, so I would appreciate more detail on this.

**Comment by FreegleBarr** [comment]
I updated the description to clarify how to reproduce this.

So, what's happening is that manually emitted particles are incorrectly "expiring" a small amount of time(0.1 seconds?) after the first particle expires. This is unnoticeable on fast moving sporadic particles, but if you have a machine gun firing particles at a fast rate you'll see a weird flicker very once in a while when all particles suddenly become invisible.

On some testing I found this to not be actually related to the first particle, but rather some periodic trigger in the glsl shaders, but I don't remember the details now and am not very acquainted with glsl in general.
